Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
+            $Protocols = Read-Host -Prompt 'Protocols' | TrimString
+        } while ($Protocols.Split(", ").Count -gt '16')
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter any Commands or aliases to run the package. For example: msedge (Max 16)'
+            $Commands = Read-Host -Prompt 'Commands' | TrimString
+        } while ($Commands.Split(", ").Count -gt '16')
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Installer Scope. machine or user'
+            $Scope = Read-Host -Prompt 'Scope' | TrimString
+        } until ([string]::IsNullOrWhiteSpace($Scope) -or ($Scope -eq 'machine' -or $Scope -eq 'user'))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Moniker (friendly name/alias). For example: vscode'
+            $Moniker = Read-Host -Prompt 'Moniker' | TrimString
+        } while ($Moniker.Length -gt 40)
+
+        #        DefaultLocale Manifest      #
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Publisher Url'
+            $PublisherUrl = Read-Host -Prompt 'Publisher Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PublisherUrl) -and ($PublisherUrl.Length -lt 5 -or $LicenseUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Publisher Support Url'
+            $PublisherSupportUrl = Read-Host -Prompt 'Publisher Support Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PublisherSupportUrl) -and ($PublisherSupportUrl.Length -lt 5 -or $PublisherSupportUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Privacy Url'
+            $PrivacyUrl = Read-Host -Prompt 'Privacy Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PrivacyUrl) -and ($PrivacyUrl.Length -lt 5 -or $PrivacyUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Application Author'
+            $Author = Read-Host -Prompt 'Author' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($Author) -and ($Author.Length -lt 1 -or $Author.Length -gt 256))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Url to the homepage of the application.'
+            $PackageUrl = Read-Host -Prompt 'Homepage' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PackageUrl) -and ($PackageUrl.Length -lt 5 -or $PackageUrl.Length -gt 2000))
+
+        while ([string]::IsNullOrWhiteSpace($License) -or $License.Length -ge 512) {
```

Comment:
why are some whiles and others do/whiles?

Reply:
all the Do's are Optional (you can press space if you don't know the value) and all the While is required for the manifest to be approved